onnxruntime/core/graph/graph_utils.cc

```diff
@@ -191,51 +193,132 @@ static void UpdateImplicitInputNameInSubgraph(Node& node,
   }
 }
 
-bool RemoveSingleInSingleOutNode(Graph& graph, Node& node) {
-  if (!IsSingleInSingleOutNode(node)) {
-    return false;
+/** Returns a vector of the output GraphEdges of a node in a given graph. */
+static std::vector<GraphEdge> GetNodeOutputEdges(Graph& graph, Node& node) {
+  std::vector<GraphEdge> output_edges;
+  for (auto it = node.OutputEdgesBegin(), end = node.OutputEdgesEnd(); it != end; ++it) {
+    output_edges.push_back(GraphEdge{node.Index(),
+                                     (*it).GetNode().Index(),
+                                     (*it).GetSrcArgIndex(),
+                                     (*it).GetDstArgIndex(),
+                                     GetNodeOutputName(node, (*it).GetSrcArgIndex())});
```

Comment:
If this is going to be a common way to construct a GraphEdge, maybe add a GraphEdge ctor that takes const Node&, const EdgeConstIterator&

Reply:
I added a new constructor (note: had to differentiate based on whether it is an incoming or an outgoing EdgeEnd).